shtdlib.sh

```diff
@@ -640,12 +640,17 @@ function finalize_path {
 }
 
 # Store full path to this script
-script_full_path="${0}"
-if [ ! -f "${script_full_path}" ] ; then
+if [[ "${BASH_SOURCE[0]}" == "${0}" ]] ; then
```

Comment:
This isn't technically the same evaluation, how about symlinks and other file like objects?